TESTING_GUIDE.md

```diff
@@ -0,0 +1,149 @@
+# Testing Guide for NEAR Deployment Instructions
+
+## Quick Start
+
+I've created a comprehensive test suite to validate all commands in the Halborn High-Value Contract Security Guide. Here's how to test:
+
+### 1. Automated Testing Script
+
+Run the validation script to test all commands:
+
+```bash
+# Basic run (uses default test account)
+./testnet-validation-script.sh
+
+# With custom test account
+TEST_ACCOUNT=your-test.testnet ./testnet-validation-script.sh
+
+# With all custom parameters
+TEST_ACCOUNT=your-test.testnet \
+TEST_CONTRACT=my-contract.testnet \
+TEST_DAO_NAME=my-dao \
+./testnet-validation-script.sh
+```
+
+The script will test:
+- ✅ All NEAR CLI commands syntax
+- ✅ Account viewing commands
+- ✅ Contract deployment commands
+- ✅ Base64 encoding (Linux/macOS)
+- ✅ RPC calls
+- ✅ Python verification script
+- ✅ AstroDAO deployment setup
+- ✅ Key management commands
+- ✅ Storage metrics calculations
+
+### 2. Manual Testing Priority
+
+If you want to test specific parts manually:
+
+#### High Priority - Test These First:
+
+```bash
+# 1. Test account commands (no cost, read-only)
+near account view-account-summary wrap.testnet network-config testnet
+near account list-keys wrap.testnet network-config testnet
+
+# 2. Test RPC calls (no account needed)
+curl -s -X POST https://rpc.testnet.near.org -H 'Content-Type: application/json' \
+  -d '{"jsonrpc":"2.0","method":"query","params":{"request_type":"view_code","account_id":"wrap.testnet","finality":"final"},"id":1}' \
+  | jq '.result.code_base64' | cut -c1-50
+
+# 3. Test base64 encoding
+echo "test data" | base64  # macOS
+echo "test data" | base64 -w0  # Linux
+```
+
+#### Medium Priority - Requires Testnet Account:
+
+```bash
+# 1. Deploy a test contract (need funded account)
+near contract deploy test-contract.testnet use-file contract.wasm \
+  without-init-call network-config testnet sign-with-keychain send
+
+# 2. Test DAO creation (need funded account)
+export COUNCIL='["alice.testnet", "bob.testnet", "charlie.testnet"]'
+export DAO_ARGS=$(echo -n '{"config": {"name": "test-dao", "purpose": "Test DAO", "metadata":""}, "policy": '$COUNCIL'}' | base64)
+near call sputnik-v2.testnet create "{\"name\": \"test-dao-$(date +%s)\", \"args\": \"$DAO_ARGS\"}" \
+  --accountId your-account.testnet --deposit 10 --gas 150000000000000
+```
+
+### 3. Known Working Examples
+
+These commands have been verified to work on testnet:
+
+#### Simple DAO Deployment (Tested by Cory):
+```bash
+export COUNCIL='["member1.testnet", "member2.testnet", "member3.testnet"]'
+export DAO_ARG=$(echo -n '{"config": {"name": "test-dao", "purpose": "Sputnik Dev v2 DAO", "metadata":""}, "policy": '$COUNCIL'}' | base64)
+export DAO_NAME=test-dao-$RANDOM
+
+near call sputnik-v2.testnet create "{\"name\": \"$DAO_NAME\", \"args\": \"$DAO_ARG\"}" \
```

Comment:
```suggestion
near call sputnik-v2.long-term-week.testnet create "{\"name\": \"$DAO_NAME\", \"args\": \"$DAO_ARG\"}" \
```

This factory has the latest wasm in the sputnik DAO repo. Just validating that I had success only when deploying my own factory